connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -216,6 +223,12 @@ protected void poll(long timeoutMs) {
             deliverMessages();
         } catch (WakeupException we) {
             log.trace("{} consumer woken up", id);
+
+            if (isPaused()) {
```

Comment:
I agree it reads a little funny. Let me see if that works with WorkerSourceTask as well.